src/pymor/models/neural_network.py

```diff
@@ -76,6 +78,60 @@ def _compute_solution(self, mu=None, **kwargs):
 
             return U
 
+    class NeuralNetworkOutputModel(Model):
+        """Class for models of the output of stationary problems that use ANNs.
+
+        This class implements a |Model| that uses a neural network for solving for the output
+        quantity.
+
+        Parameters
+        ----------
+        neural_network
+            The neural network that approximates the mapping from parameter space
+            to output space. Should be an instance of
+            :class:`~pymor.models.neural_network.FullyConnectedNN` with input size that
+            matches the (total) number of parameters and output size equal to the
+            dimension of the output space.
+        parameters
+            |Parameters| of the reduced order model (the same as used in the full-order
+            model).
+        products
+            A dict of inner product |Operators| defined on the discrete space the
+            problem is posed on. For each product with key `'x'` a corresponding
+            attribute `x_product`, as well as a norm method `x_norm` is added to
+            the model.
+        error_estimator
+            An error estimator for the problem. This can be any object with
+            an `estimate_error(U, mu, m)` method. If `error_estimator` is
+            not `None`, an `estimate_error(U, mu)` method is added to the
+            model which will call `error_estimator.estimate_error(U, mu, self)`.
+        visualizer
+            A visualizer for the problem. This can be any object with
+            a `visualize(U, m, ...)` method. If `visualizer`
+            is not `None`, a `visualize(U, *args, **kwargs)` method is added
+            to the model which forwards its arguments to the
+            visualizer's `visualize` method.
+        name
+            Name of the model.
+        """
+
+        def __init__(self, neural_network, parameters={}, products=None, error_estimator=None,
+                     visualizer=None, name=None):
```

Comment:
```suggestion
        def __init__(self, neural_network, parameters={}, error_estimator=None, name=None):
```

I don't think it makes sense to have these additional quantities as there is no state space they could work with.

Reply:
Done in 3e25bb7.